src/Form/ImportantSolrSettingsForm.php

```diff
@@ -0,0 +1,249 @@
+
+    return $fields_by_index;
+  }
+  
+  /**
+   * AJAX callback function when user selects Solr Server
+   * 
+   * Updates both Index and Solr Field dropdowns in the returned AjaxResponse
+   */
+  public function onServerSelect(array &$form, FormStateInterface $form_state) {
+    // Pass the selected server to get its indexes
+    $indexes_by_server = $this->getIndexes($form_state->getValue('server_select'));
+    // As a default for this new selection, use the first index the array and fetch its fields
+    $fields = $this->getSolrFields(array_key_first($indexes_by_server));
+    
+    // Set the new #options for Index
+    $form['type']['index_select']['#options'] = $indexes_by_server;
+    // Set the new #options for Solr Field
+    $form['type']['solr_field_select']['#options'] = $fields;
+
+    // Construct our own response to return/change multiple form fields
+    $response = new AjaxResponse();
+    $response->addCommand(new ReplaceCommand("#index-select", ($form['type']['index_select'])));
+    $response->addCommand(new ReplaceCommand("#solr-field-select", ($form['type']['solr_field_select'])));
+    
+    return $response;
+   }
+
+  /**
+   * AJAX callback function when user selects Index
+   *
+   * Updates Solr Field dropdown
+   */
+  public function onIndexSelect(array &$form, FormStateInterface $form_state) {
+    // Get solr fields for selected Index
+    $fields = $this->getSolrFields($form_state->getValue('index_select'));
+    $form['type']['solr_field_select']['#options'] = $fields;
+    
+    return $form['type']['solr_field_select'];  
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('strawberryfield.archipelago_solr_settings');
+    
+    // Get all Solr servers
+    $servers = $this->entityTypeManager
+      ->getStorage('search_api_server')
+      ->loadMultiple();
+
+    // Create a fieldset for the Type setting
+    $form['type'] = array(
+      '#type' => 'fieldset',
+      '#title' => $this->t('Solr Field for Type in View Mode mapping'),
+      '#markup' => '<p>Choose the Solr field that will be used in our ADO to View Mode mapper. <br><br> Fields are specific to their Solr Server and Index. <br> To edit servers, indexes, and fields, visit <a href=" /admin/config/search/search-api/">Search API Config</a><br> For info and settings about the View Mode mapper, visit the <a href="/admin/config/archipelago/viewmode_mapping">ADO to View Mode Config</a></p>'
+    );
+
+    $form['type']['server_select'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Step 1. Select Server'),
+      '#options' => $this->formatOptions($servers),
+      '#default_value' => $config->get('type_server'),
+      '#ajax' => [
+        'callback' => '::onServerSelect',
+        'disable-refocus' => FALSE,
+        'event' => 'change',
+        'wrapper' => 'index-select', 
+      ]
+    ];
+    
+    $form['type']['index_select'] = [
```

Comment:
In Step2 and Step3 you can use #states too. When i said #states where not the right way i meant for populating the values. The way you do it via Ajax is the right now, but still, it would be great that in case of no values of Step1, step2 would be not shown, same goes with step3 if step2 is empty.